lib/features/geo/cubit/manage_geo_cubit.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:geolocator/geolocator.dart';
+import 'package:refuelety/api/api.dart';
+import 'package:refuelety/core/app_service_locator.dart';
+
+part 'manage_geo_cubit.freezed.dart';
+part 'manage_geo_state.dart';
+
+class ManageGeoCubit extends Cubit<ManageGeoState> {
+  ManageGeoCubit() : super(const ManageGeoState.initial());
+
+  Future<void> checkPermission() async {
+    LocationPermission permission = await Geolocator.checkPermission();
+    final bool isServiceEnabled = await Geolocator.isLocationServiceEnabled();
+
+    if (!isServiceEnabled) {
+      permission = await Geolocator.requestPermission();
+    } else if (permission == LocationPermission.denied) {
+      permission = await Geolocator.requestPermission();
+    }
+    if (permission == LocationPermission.deniedForever) {
+      emit(const ManageGeoState.permissionDenied());
+    }
+  }
```

Comment:
This belongs to the `OnlineGeoService`.  So you can check the checkPermission() and if false, than you can emit permissionDenied